### PR TITLE
channelmixerrgb: when creating new instances avoid two CAT.

### PR DESCRIPTION
**Pull request by TurboGit**
When an instance of the channelmixerrgb already exists and is
already doing CAT set the new instance adpatation to none.

Will make easier working with multiple instance of channelmixerrgb.

### CODE REVIEW

File: src/iop/channelmixerrgb.c [hunk 1]
```diff
@@ -1880,6 +1880,29 @@ void reload_defaults(dt_iop_module_t *module)
...
+
+  GList *iop = darktable.develop->iop;
+  while(iop)
+  {
+    const dt_iop_module_t *m = (dt_iop_module_t *)iop->data;
+    if(module != m && m->enabled && !strcmp(m->op, "channelmixerrgb"))
+    {
+      const dt_iop_channelmixer_rgb_params_t *mp =
+        (dt_iop_channelmixer_rgb_params_t *)m->params;
+      if(mp->adaptation != DT_ADAPTATION_RGB)
```
Comment (aurelienpierre) [inline src/iop/channelmixerrgb.c:1896]: if(!(mp->adaptation == DT_ADAPTATION_RGB || mp->illuminant == DT_ILLUMINANT_PIPE))

**Review by aurelienpierre — CHANGES_REQUESTED** [review]

**Comment by aurelienpierre** [comment]
OOooooh so we can fetch one module's parameters from another module like that ?

**Comment by TurboGit** [comment]
> OOooooh so we can fetch one module's parameters from another module like that ?

Yes We Can !

File: src/iop/channelmixerrgb.c (hunk 1 above)
Comment (TurboGit) [reply src/iop/channelmixerrgb.c]: Done.

**Comment by dterrahe** [comment]
Does this make the "2nd" instance permanently behave different from the "1st"? What about if the ordering is changed? And after saving/reloading/switching images?

**Comment by TurboGit** [comment]
> Does this make the "2nd" instance permanently behave different from the "1st"? What about if the ordering is changed? And after saving/reloading/switching images?

There is no assumption about instance number. Only if one of the active instance is having CAT enabled then the newly created one won't have it. If a single instance exists and does not have CAT the new instance will have CAT.

**Comment by johnny-bit** [comment]
This seems to also allow multiple instances WITH CAT, just that the default for new instances is bypass :)

**Comment by matt-maguire** [comment]
Having multiple instances of channelmixerrgb with CAT is a valid use case (eg. when used together with masks). If one does this, there is a warning provided, however the software shouldn't actively interfere which what the user is trying to do.

**Comment by chhil** [comment]
May want to document this behavior. 

@matt-maguire did mention in another thread that people may want to use masking and still use white balance and in that case they may not want to bypass it.

I am happy with this bypass on second module, already got bit by it.

**Comment by TurboGit** [comment]
Yes all keep possible, yet by default (the expected more common usage) is that only one instance has CAT.

**Comment by TurboGit** [comment]
> I am happy with this bypass on second module, already got bit by it.

Yes, I've done that because I saw your message.

**Pull request merged**